src/components/AccountIcon.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+import React, { Component, PropTypes } from 'react'
+import { Image } from 'react-native'
+import { blockiesIcon } from '../util/native'
+
+async function displayIcon (self, seed) {
+  try {
+    let icon = await blockiesIcon(seed)
+    self.setState({
+      icon: icon
+    })
+  } catch (e) {
+    console.log(e)
+  }
+}
+
+export default class AccountIcon extends Component {
+  static propTypes = {
+    seed: PropTypes.string.isRequired
+  }
+
+  state = {}
+
+  componentDidMount () {
+    displayIcon(this, this.props.seed)
+  }
+
+  componentWillReceiveProps (newProps) {
+    displayIcon(this, newProps.seed)
```

Comment:
This can be fired even if `seed` was not changed, so it's better to early exit if there are no changes.